base/src/main/java/proguard/optimize/inline/LocalUsageRemover.java

```diff
@@ -0,0 +1,56 @@
+package proguard.optimize.inline;
+
+import proguard.classfile.Clazz;
+import proguard.classfile.Method;
+import proguard.classfile.ProgramClass;
+import proguard.classfile.ProgramMethod;
+import proguard.classfile.attribute.Attribute;
+import proguard.classfile.attribute.CodeAttribute;
+import proguard.classfile.attribute.visitor.AllAttributeVisitor;
+import proguard.classfile.attribute.visitor.AttributeVisitor;
+import proguard.classfile.editor.CodeAttributeEditor;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.instruction.VariableInstruction;
+import proguard.classfile.instruction.visitor.InstructionVisitor;
+import proguard.classfile.visitor.MemberVisitor;
+
+public class LocalUsageRemover implements MemberVisitor {
+    private final CodeAttributeEditor codeAttributeEditor;
+    private final int argumentIndex;
+
+    public LocalUsageRemover(CodeAttributeEditor codeAttributeEditor, int argumentIndex) {
+        this.codeAttributeEditor = codeAttributeEditor;
+        this.argumentIndex = argumentIndex;
+    }
+
+    @Override
+    public void visitProgramMethod(ProgramClass programClass, ProgramMethod programMethod) {
+        programMethod.accept(programClass, new AllAttributeVisitor(new AttributeVisitor() {
+            @Override
+            public void visitAnyAttribute(Clazz clazz, Attribute attribute) {}
```

Comment:
Nested visitors again, please clean up by making LocalUsageRemove implement InstructionVisitor and have the class implement the methods instead of implementing them in the arguments.